### DISCUSSION TITLE
[Idea] backoffice authentification with 2FA

### DISCUSSION THREAD

**Discussion by jf-viguier** [PrestaShop/PrestaShop#31024] [Ideas]
### The solution

Implement a 2FA auth for the backoffice or a way to plug one.

### Do you plan to work on this subject?

- [ ] I'm willing to contribute a [formal specification](https://github.com/PrestaShop/prestashop-specs).
- [ ] I'm willing to provide any wireframes or design assets required for this feature.
- [ ] I'm willing to submit a Pull Request that implements this feature.
- [ ] I'm willing to help verify that the implemented feature works as intended and produces no unintended side effects.

**Comment by yucom31** [comment]
Or propose Identity Federation proposing MFA or more cybercontrols on Authentication > Google / MS/Github/FB (?)

**Reply by touchweb-vincent** [reply]
Yubikey (OTP) and Google Authenticator (TOTP) could be a nice start.
https://github.com/enygma/yubikey / https://github.com/PHPGangsta/GoogleAuthenticator

**Comment by matks** [comment]
Similar to https://github.com/PrestaShop/PrestaShop/issues/23243
https://github.com/PrestaShop/PrestaShop/issues/25601
